relay/messages/message.go

```diff
@@ -55,37 +59,53 @@ type HelloResponse struct {
 	InstanceAddress string
 }
 
-// DetermineClientMsgType determines the message type from the first byte of the message
-func DetermineClientMsgType(msg []byte) (MsgType, error) {
+// ValidateVersion checks if the given version is supported by the protocol
+func ValidateVersion(msg []byte) (int, error) {
+	if len(msg) < 1 {
+		return 0, ErrInvalidMessageLength
+	}
+	version := int(msg[0])
+	if version != CurrentProtocolVersion {
+		return 0, fmt.Errorf("%d: %w", version, ErrUnsupportedVersion)
+	}
+	return version, nil
+}
+
+// DetermineClientMessageType determines the message type from the first the message
+func DetermineClientMessageType(msg []byte) (MsgType, error) {
+	if len(msg) < 1 {
+		return 0, ErrInvalidMessageLength
+	}
+
 	msgType := MsgType(msg[0])
 	switch msgType {
-	case MsgTypeHello:
-		return msgType, nil
-	case MsgTypeTransport:
-		return msgType, nil
-	case MsgTypeClose:
-		return msgType, nil
-	case MsgTypeHealthCheck:
+	case
+		MsgTypeHello,
+		MsgTypeTransport,
+		MsgTypeClose,
+		MsgTypeHealthCheck:
 		return msgType, nil
 	default:
-		return 0, fmt.Errorf("invalid msg type, len: %d", len(msg))
+		return 0, fmt.Errorf("invalid msg type %d, len: %d", msgType, len(msg))
```

Comment:
The len(msg) is invalid because this function get the buf[1:n]

Reply:
fixed